Mac full screen media controls do no match AVKit floating controls

Phinny01 opened the pull request:
#### faf60369ae617f3b553540c9dbaa5939de0dc1cc
<pre>
Mac full screen media controls do no match AVKit floating controls
<a href="https://bugs.webkit.org/show_bug.cgi?id=301025">https://bugs.webkit.org/show_bug.cgi?id=301025</a>
<a href="https://rdar.apple.com/162909216">rdar://162909216</a>

Reviewed by Andy Estes.

Update macOS inline media controls to match AVKit&apos;s floating controls design. Increase bottom
control bar height and slider heights. Replace circular and bar shaped slider thumbs with
pill-shaped design for both time control scrubber and volume slider.

* LayoutTests/media/modern-media-controls/macos-fullscreen-media-controls/macos-fullscreen-media-controls-drag-is-prevented-over-button-expected.txt:
* LayoutTests/media/modern-media-controls/macos-fullscreen-media-controls/macos-fullscreen-media-controls-drag-is-prevented-over-button.html:
* LayoutTests/platform/mac-sonoma-wk1/imported/w3c/web-platform-tests/media-source/mediasource-seek-during-pending-seek-expected.txt:
* Source/WebCore/Modules/modern-media-controls/controls/macos-fullscreen-media-controls.css:
(.media-controls.mac.fullscreen .buttons-container.left):
(.media-controls.mac.fullscreen .buttons-container.left .slider):
* Source/WebCore/Modules/modern-media-controls/controls/macos-fullscreen-media-controls.js:
(MacOSFullscreenMediaControls.prototype._handleMousedown):
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::macOSFullscreenMediaControlsStyleSheet):
(WebCore::RenderThemeCocoa::mediaControlsStyleSheets):

Canonical link: <a href="https://commits.webkit.org/303335@main">https://commits.webkit.org/303335@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ff96b583f39a26708b5691b646968689c0880bdf

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132021 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4513 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43037 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139535 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83928 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/fba95830-0b9d-4538-bca6-269ea2127d44) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133891 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4456 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4275 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100921 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68299 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7d3f7838-6e98-449e-a229-e722d155247c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134967 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/3171 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118239 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81711 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/4f7e3921-0203-4999-b8f4-cd5da644897a) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/3066 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/939 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82755 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111821 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36358 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142182 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4183 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36940 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109290 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4264 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3644 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109462 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27740 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3173 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114516 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/57404 "The change is no longer eligible for processing.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4237 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32910 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4068 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67683 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4328 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4196 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebCore/Modules/modern-media-controls/controls/macos-fullscreen-media-controls.css:
```diff
@@ -54,6 +54,7 @@
 .media-controls.mac.fullscreen .buttons-container.left {
     top: 16px;
     height: 16px;
+    width: 100px !important;
```
Does this change the appearance of the fullscreen controls on platforms that don't support Liquid Glass?

Phinny01 replied:
No it doesn't since I set the slider width in the css file now,  I needed to set the width of the .buttons-container.left to accommodate the slider width else the slider becomes smaller than it should be.

aestes commented on Source/WebCore/Modules/modern-media-controls/controls/macos-fullscreen-media-controls.css:
```diff
@@ -74,6 +75,9 @@
     top: 18px;
 }
 
+.media-controls.mac.fullscreen .buttons-container.left .slider {
+    width: 60px;
+}
```
Does this change the appearance of the fullscreen controls on platforms that don't support Liquid Glass?

Phinny01 replied:
This was in macos-fullscreen-media-controls.js earlier but I moved it to css to avoid using the controls host in the .js file

aestes commented on Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
```diff
@@ -555,8 +647,10 @@ static inline bool canShowCapsLockIndicator()
         mediaControlsStyleSheets.append(glassMaterialMediaControlsStyleSheet());
 
-    if (mediaElement.document().settings().mediaControlsMacInlineSizeSpecsEnabled())
+    if (mediaElement.document().settings().mediaControlsMacInlineSizeSpecsEnabled()) {
```
This is OK as-is, although `mediaControlsMacInlineSizeSpecsEnabled` is now slightly misnamed.

Phinny01 replied:
I agree. I can do that in a follow up patch

Commit:
Mac full screen media controls do no match AVKit floating controls
https://bugs.webkit.org/show_bug.cgi?id=301025
rdar://162909216

Reviewed by Andy Estes.

Update macOS inline media controls to match AVKit's floating controls design. Increase bottom
control bar height and slider heights. Replace circular and bar shaped slider thumbs with
pill-shaped design for both time control scrubber and volume slider.

* LayoutTests/media/modern-media-controls/macos-fullscreen-media-controls/macos-fullscreen-media-controls-drag-is-prevented-over-button-expected.txt:
* LayoutTests/media/modern-media-controls/macos-fullscreen-media-controls/macos-fullscreen-media-controls-drag-is-prevented-over-button.html:
* LayoutTests/platform/mac-sonoma-wk1/imported/w3c/web-platform-tests/media-source/mediasource-seek-during-pending-seek-expected.txt:
* Source/WebCore/Modules/modern-media-controls/controls/macos-fullscreen-media-controls.css:
(.media-controls.mac.fullscreen .buttons-container.left):
(.media-controls.mac.fullscreen .buttons-container.left .slider):
* Source/WebCore/Modules/modern-media-controls/controls/macos-fullscreen-media-controls.js:
(MacOSFullscreenMediaControls.prototype._handleMousedown):
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::macOSFullscreenMediaControlsStyleSheet):
(WebCore::RenderThemeCocoa::mediaControlsStyleSheets):

Canonical link: https://commits.webkit.org/303335@main
